client/src/js/modules/imaging/views/queuecontainer.js

```diff
@@ -692,29 +698,63 @@ define(['marionette',
             if (p) this.applyModel(p, true)
         },
 
-        applyPresetAll:function(e) {
+        applyPresetAll: async function(e) {
             e.preventDefault()
+            var self = this
 
             var p = this.plans.findWhere({ DIFFRACTIONPLANID: this.ui.preset.val() })
-            if (p) this.applyModel(p, false)
+            if (p) {
+                self.ui.applyall.html('Applying...').prop('disabled', true)
+                var promises = await this.applyModel(p, false)
+
+                if (promises && promises.length > 0) {
+                    $.when.apply($, promises).always(function() {
```

Comment:
Is it possible to rewrite this with native functions (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) instead? I can have a go at it when I have some time

Reply:
Ok, have done. Have used Promise.allSettled as Promise.all would return immediately if any one failed.